windows/OrientationWindows/OrientationWindows.cpp

```diff
@@ -0,0 +1,211 @@
+                DisplayInformation::AutoRotationPreferences(DisplayOrientations::PortraitFlipped);
+                LockDidChange("PORTRAIT-UPSIDEDOWN");
+            }
+        }
+    }
+
+    void OrientationLockerModule::LockToLandscape() noexcept {
+        UserInteractionMode mode = m_viewSettings.UserInteractionMode();
+        if (mode == UserInteractionMode::Touch) {
+            if (m_displayInfo.AutoRotationPreferences() != DisplayOrientations::Landscape) {
+                DisplayInformation::AutoRotationPreferences(DisplayOrientations::Landscape);
+                LockDidChange("LANDSCAPE");
+            }
+        }
+    }
+
+    void OrientationLockerModule::LockToLandscapeRight() noexcept {
+        UserInteractionMode mode = m_viewSettings.UserInteractionMode();
+        if (mode == UserInteractionMode::Touch) {
+            if (m_displayInfo.AutoRotationPreferences() != DisplayOrientations::LandscapeFlipped) {
+                DisplayInformation::AutoRotationPreferences(DisplayOrientations::LandscapeFlipped);
+                LockDidChange("LANDSCAPE-RIGHT");
+            }
+        }
+    }
+
+    void OrientationLockerModule::LockToLandscapeLeft() noexcept {
+        UserInteractionMode mode = m_viewSettings.UserInteractionMode();
+        if (mode == UserInteractionMode::Touch) {
+            if (m_displayInfo.AutoRotationPreferences() != DisplayOrientations::Landscape) {
+                DisplayInformation::AutoRotationPreferences(DisplayOrientations::Landscape);
+                LockDidChange("LANDSCAPE-LEFT");
+            }
+        }
+    }
+
+    void OrientationLockerModule::UnlockAllOrientations() noexcept {
+        UserInteractionMode mode = m_viewSettings.UserInteractionMode();
+        if (mode == UserInteractionMode::Touch) {
+            DisplayOrientations allOrientations = DisplayOrientations::Landscape | DisplayOrientations::LandscapeFlipped | DisplayOrientations::Portrait | DisplayOrientations::PortraitFlipped;
+            if (m_displayInfo.AutoRotationPreferences() != allOrientations) {
+                DisplayInformation::AutoRotationPreferences(allOrientations);
+                GetOrientation(LockDidChange);
+            }
+        }
+    }
+
+    void OrientationLockerModule::GetConstants(React::ReactConstantProvider& provider) noexcept {
+        if (!m_context.UIDispatcher().HasThreadAccess()) {
+            HANDLE ghSetConstantsEvent;
+            ghSetConstantsEvent = CreateEvent(
+                NULL,                       // default security attributes
+                TRUE,                       // manual-reset event
+                FALSE,                      // initial state is nonsignaled
+                TEXT("SetConstantsEvent")   // object name
+            );
+            if (ghSetConstantsEvent == NULL)
+            {
+                printf("CreateEvent for ghSetConstantsEvent failed (%d)\n", GetLastError());
```

Comment:
probably want to assert instead. If the event wasn't created we should probably just crash